### PR TITLE
C4 m0hamed s12 api

### PR DESCRIPTION
**Pull request by m0hamed**
#363

### CODE REVIEW

File: arability/app/controllers/api_controller.rb [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+  #   creates the api and redirects to project page
+  # Failure:
+  #   shows the form again to check validations
+  def create
+    @api_key = ApiKey.new(params[:api_key])
+    @api_key.developer = current_developer
+
+    if @api_key.save
+      redirect_to api_keys_list_path,
+        flash: {success: "Your API key has been successfully created" }
```
Comment (NourhanMohamed) [inline arability/app/controllers/api_controller.rb]: use a space after the {

File: arability/app/controllers/api_controller.rb [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+  # Success:
+  #   deletes the api and returns status code 200 success
+  # Failure:
+  #   returns status code 403 forbidden
+  def delete
+    api_key = ApiKey.find_by_id(params[:api_key_id])
+
+    if api_key && api_key.developer == current_developer
+      api_key.destroy
+      render nothing: true, :status => 200
```
Comment (NourhanMohamed) [inline arability/app/controllers/api_controller.rb]: remove =>

File: arability/app/controllers/api_controller.rb [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+  # Params:
+  #   --
+  # Success:
+  #   Displays the help pages
+  # Failure:
+  #   --
+  def help
+  end
+
+  def test
```
Comment (NourhanMohamed) [inline arability/app/controllers/api_controller.rb:177]: what does that do?

File: arability/app/controllers/api_controller.rb [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+  #   text: a string of words
+  # Success:
+  #   returns a string of words that are capitalized
+  # Failure:
+  #   --
+  def capitalize_sentence text
+    words = text.split(" ").map { |word| word.capitalize }
+    words.join(" ")
+  end
+
```
Comment (NourhanMohamed) [inline arability/app/controllers/api_controller.rb]: remove extra line

File: arability/app/models/api_key.rb [hunk 5]
```diff
@@ -0,0 +1,86 @@
+class ApiKey < ActiveRecord::Base
+
```
Comment (NourhanMohamed) [inline arability/app/models/api_key.rb:2]: remove extra line

File: arability/app/models/api_key.rb [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+  #   returns the best synonyms for the passed keyword given the parameters
+  # Failure:
+  #   returns nil if the keyword doesnt exist or no synonyms are found for it
+  def get_synonym_for(keyword, country = nil, age_from = nil, age_to = nil,
+        gender = nil, education = nil)
+    keyword = Keyword.find_by_name(keyword)
+    return nil if keyword.blank?
+
+    if self.project.present?
+
```
Comment (NourhanMohamed) [inline arability/app/models/api_key.rb:38]: remove extra line

File: arability/app/models/api_key.rb [hunk 7]
```diff
@@ -0,0 +1,86 @@
...
+  # Params:
+  #   --
+  # Success:
+  #   the website gets downcased
+  # Failure:
+  #   --
+  def downcase_website
+    self.website.downcase! if self.website.present?
+  end
+
```
Comment (NourhanMohamed) [inline arability/app/models/api_key.rb]: remove extra line

File: arability/app/controllers/api_controller.rb [hunk 8]
```diff
@@ -0,0 +1,195 @@
...
+  #   Displays the help pages
+  # Failure:
+  #   --
+  def help
+  end
+
+  def test
+  end
+
+private
```
Comment (NourhanMohamed) [inline arability/app/controllers/api_controller.rb:180]: leave a new line after the private keyword

File: arability/app/views/api/help.html.erb [hunk 9]
```diff
@@ -0,0 +1,85 @@
+<%= link_to "<< #{t :back_to_api }", api_keys_list_path %>
```
Comment (NourhanMohamed) [inline arability/app/views/api/help.html.erb]: leave a space after the t

File: arability/app/views/api/help.html.erb [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+      The education level of the user. The valid values are "School", "University",
+      "Graduate"
+    </td>
+  </tr>
+  <tr>
+    <td>
+      age_from, age_to
+    </td>
+    <td>
+      The age range to target. The maximum range is 10-100 years old.
```
Comment (NourhanMohamed) [inline arability/app/views/api/help.html.erb:94]: I suggest you remove the word maximum .. it's a range anyways "this is IMHO, not reviewing =)"

File: arability/app/views/api/test.html.erb [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+    <div>
+      <input class="btn btn-large btn-primary pull-right" name="commit"
+        data-arabize="sign in" style="margin-top:-7px; padding-top:4px; padding-bottom:4px;"
+        type="submit" value="Sign In" />
+    </div>
+  </form>
+</span>
+<script>
+  <%= render "script.js" %>
+  $.arabize({api_key: "40dd01f1c897dedfe7e9888c76f56117"})
```
Comment (NourhanMohamed) [inline arability/app/views/api/test.html.erb]: leave a space after { and before }
This file was for testing purposes, ri8? I mean for the demo?

File: arability/app/controllers/api_controller.rb (hunk 3 above)
Comment (m0hamed) [inline arability/app/controllers/api_controller.rb:177]: this displays a view that has my api so as to show it at the demo.

File: arability/app/models/api_key.rb (hunk 5 above)
Comment (m0hamed) [inline arability/app/models/api_key.rb:2]: these lines are actually there to make the code more readable

File: arability/app/views/api/test.html.erb (hunk 11 above)
Comment (m0hamed) [inline arability/app/views/api/test.html.erb]: yes

File: arability/app/views/api/test.html.erb [hunk 12]
```diff
@@ -0,0 +1,25 @@
+<span style="direction: rtl">
+  <form accept-charset="UTF-8" action="/en/gamers/sign_in" class="form-signin"
```
Comment (NourhanMohamed) [inline arability/app/views/api/test.html.erb:2]: use ruby path helpers

File: arability/app/views/api/index.html.erb [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+      <ul class="unstyled">
+        <% @api_key.errors.full_messages.each do |msg| %>
+          <li><%= msg %></li>
+        <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <%= form_for @api_key, url: { action: "create" },
+    html: {:class => "form-horizontal"}, method: :post do |f| %>
```
Comment (NourhanMohamed) [inline arability/app/views/api/index.html.erb]: add a space after { and before }

File: arability/app/views/api/test.html.erb (hunk 12 above)
Comment (m0hamed) [inline arability/app/views/api/test.html.erb:2]: this is the test for the api it should be in its own file outside the server

**Pull request merged**